component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/dao/JWTStorageManager.java

```diff
@@ -26,156 +26,192 @@
 import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants;
-import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.SQLQueries;
 import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.internal.JWTServiceDataHolder;
+import org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.util.Util;
 
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Timestamp;
+import java.util.ArrayList;
 import java.util.Calendar;
+import java.util.List;
 import java.util.TimeZone;
 
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.DEFAULT_TENANT_ID;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.GET_JWT;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.GET_JWT_DETAILS;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.INSERT_JWD_ID;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.SQLQueries.EXP_TIME;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.SQLQueries.TENANT_ID;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.SQLQueries.TIME_CREATED;
+
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isDB2DB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isH2DB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isMSSqlDB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isMariaDB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isMySQLDB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isOracleDB;
 import static org.wso2.carbon.identity.core.util.JdbcUtils.isPostgreSQLDB;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.UPSERT_H2;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.UPSERT_MSSQL_DB2;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.UPSERT_MYSQL;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.UPSERT_ORACLE;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.UPSERT_POSTGRESQL;
 
 /**
  * JWT token persistence is managed by JWTStorageManager
- * It saved JWTEntry instances in Identity Database.
+ * It saved JWTEntry instances in IDN_OIDC_JTI table of Identity Database.
  */
 public class JWTStorageManager {
 
     private static final Log log = LogFactory.getLog(JWTStorageManager.class);
 
     /**
-     * Check whether a JWT Entry with given jti exists in the DB.
-     *
-     * @param jti JWT token id
-     * @return true if an entry is found
-     * @throws IdentityOAuth2Exception when exception occurs
-     */
-    public boolean isJTIExistsInDB(String jti) throws OAuthClientAuthnException {
```

Comment:
Is it safe to remove a public method?

Reply:
Yes.. This public method exists in the DAO layer. No other component going to use it. Hence I remove it.

Reply:
Is it possible to customer using this method in his custom implementation.

Reply:
Also shall we check whether is is okay to change public method signatures.

Reply:
Ack